addon/components/bs-button.d.ts

```diff
@@ -0,0 +1,54 @@
+import Component from '@glimmer/component';
+
+export type BsButtonType = 'primary' | 'secondary' | 'success' | 'danger' | 'warning' | 'info' | 'light' | 'dark';
```

Comment:
Valid button types depend on the Bootstrap theme being used. They are derived values from the [color maps](https://getbootstrap.com/docs/5.3/customize/color/#color-sass-maps). In the case of `@type` argument of `<BsButton>` the allowed values are the entries in [`$theme-colors` SASS map](https://github.com/twbs/bootstrap/blob/aff8f0dc9ae0ca34f4239c9395433179d1438fe6/scss/_variables.scss#L312-L321).

Not sure how we best deal with it in Glint type. I see three options:

1. Typing those arguments as `string` rather than using an enum. This has the major trade-off that the types are by far less usable.
2. Typing those arguments as an enums matching Bootstrap default values but allowing consumers to override it somehow. Not sure how that would work technically to be honest.
3. Deriving the correct type from Bootstrap SASS theme being used. I have no idea if that's technical possible and if the required tooling exists already.

I feel wrong types are worse than no types. I would tend towards typing them as `string` unless we have an option to override or derive from Bootstrap theme. But I maybe opinionated as we customize `$theme-colors` SASS map at work.

I fear there are many more cases in which the type depends on the Bootstrap theme being used. So maybe we could use the `@type` argument of `<BsButton>` just as a showcase for that general topic.

Reply:
I know, I think I generally typed the places where this is used as BsButtonType | string, just to have some default suggestions, but to allow any string anyway. Feel free to change it however you like though.

Reply:
So I tried this again, and it turns out that once there's `| string` in an union type, TS can't show default suggestions and widens the type to just `string`.
I will probably remove this type and put it as string. I don't think there's any way to infer this, definitely not from SASS.

As for wrong types being worse than no types, in general I would say any types are better than no types - there is no way to type an addon of this size correctly on a first try, but as long as it doesn't modify the runtime behavior, it doesn't matter, it's just a productivity boost, and some types will be better than no types People will refine them over time for sure.

Reply:
> As for wrong types being worse than no types, in general I would say any types are better than no types - there is no way to type an addon of this size correctly on a first try, but as long as it doesn't modify the runtime behavior, it doesn't matter, it's just a productivity boost, and some types will be better than no types. People will refine them over time for sure anyway.

Fully agree. My statement was not about having wrong types by mistake. Rather than if we should include types, which we know to be wrong in some cases.

Reply:
> So I tried this again, and it turns out that once there's `| string` in an union type, TS can't show default suggestions and widens the type to just `string`.
> I will probably remove this type and put it as string. I don't think there's any way to infer this, definitely not from SASS.

I feel in that case typing as `string` at least in first iteration seems to be the best. We could make the type more precise while still being correct in all cases in a later iteration.

Reply:
You're right, trying to provide default values doesn't bring any value for now, so I replaced the BsButtonType for string everywhere